gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/InstrumentatedGoogleHadoopFileSystem.java

```diff
@@ -321,10 +326,35 @@ public GhfsStorageStatistics getStorageStatistics() {
    */
   @Override
   public IOStatistics getIOStatistics() {
+    setHttpStatistics();
     return instrumentation != null ? instrumentation.getIOStatistics() : null;
   }
 
   public GhfsInstrumentation getInstrumentation() {
     return this.instrumentation;
   }
+
+  /** Set the Value for http get and head request related statistics keys */
+  private void setHttpStatistics() {
+    try {
+      GoogleCloudStorage gcs = getGcsFs().getGcs();
+      incrementStatistic(
```

Comment:
we need clear the stat before incrementing to ensure we don't double count, if `getIOStatistics` is called multiple times

The method needs to be idempotent

Reply:
Updated